cnxepub/adapters.py

```diff
@@ -286,14 +286,19 @@ def __init__(self, item, package):
         self._html = etree.parse(self._item.data)
 
         metadata = parse_metadata(self._html)
-        content_xpath = (
-            "//xhtml:body/node()[not(self::node()[@data-type='metadata'])]")
-        nsmap = {'xhtml': "http://www.w3.org/1999/xhtml"}
-
-        content = io.BytesIO(
-            b''.join([
-                isinstance(n, str) and n.encode('utf-8') or etree.tostring(n)
-                for n in self._html.xpath(content_xpath, namespaces=nsmap)]))
+        body = self._html.xpath('//xhtml:body',
+                                namespaces=HTML_DOCUMENT_NAMESPACES)[0]
+        metadata_node = self._html.xpath(
+                                    "//xhtml:body/*[@data-type='metadata']",
+                                    namespaces=HTML_DOCUMENT_NAMESPACES)
+        if len(metadata_node) > 0:
```

Comment:
Hmm, I suppose there might be more than one metadata stanza, accidentally. That would be more robust. Will change.